<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Form</class>
 <widget class="QWidget" name="Form">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>443</width>
    <height>707</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="joint_groupBox">
     <property name="title">
      <string>Inputs</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="spacing">
       <number>3</number>
      </property>
      <property name="leftMargin">
       <number>3</number>
      </property>
      <property name="topMargin">
       <number>3</number>
      </property>
      <property name="rightMargin">
       <number>3</number>
      </property>
      <property name="bottomMargin">
       <number>3</number>
      </property>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_13">
        <item>
         <widget class="QLabel" name="joint_list_lable">
          <property name="text">
           <string>Points</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QListWidget" name="joint_list">
          <property name="statusTip">
           <string>Choose a point to be a revolute joint.</string>
          </property>
          <property name="editTriggers">
           <set>QAbstractItemView::NoEditTriggers</set>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QLabel" name="inputs_label_right1">
        <property name="text">
         <string>&gt;&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_15">
        <item>
         <widget class="QLabel" name="base_link_list_lable">
          <property name="text">
           <string>Base link</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QListWidget" name="base_link_list">
          <property name="statusTip">
           <string>Coordinate reference.</string>
          </property>
          <property name="editTriggers">
           <set>QAbstractItemView::NoEditTriggers</set>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QLabel" name="inputs_label_right2">
        <property name="text">
         <string>&gt;&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_16">
        <item>
         <widget class="QLabel" name="drive_link_list_lable">
          <property name="text">
           <string>Drive link</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QListWidget" name="drive_link_list">
          <property name="statusTip">
           <string>Coordinate movement reference.</string>
          </property>
          <property name="editTriggers">
           <set>QAbstractItemView::NoEditTriggers</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="variable_list_add">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="statusTip">
           <string>Add to variable list with above settings.</string>
          </property>
          <property name="icon">
           <iconset resource="../../icons.qrc">
            <normaloff>:/icons/arrow_down.png</normaloff>:/icons/arrow_down.png</iconset>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="variable_groupBox">
     <property name="title">
      <string>Variables</string>
     </property>
     <layout class="QHBoxLayout" name="inputs_dial_layout">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_10">
        <item>
         <widget class="QListWidget" name="variable_list">
          <property name="statusTip">
           <string>All the variable of this mechanism.</string>
          </property>
          <property name="editTriggers">
           <set>QAbstractItemView::NoEditTriggers</set>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QPushButton" name="variable_remove">
            <property name="statusTip">
             <string>Delete the specified variable.</string>
            </property>
            <property name="icon">
             <iconset resource="../../icons.qrc">
              <normaloff>:/icons/delete.png</normaloff>:/icons/delete.png</iconset>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="Line" name="line_7">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_18">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_4">
              <item>
               <widget class="QLabel" name="variable_speed_label">
                <property name="text">
                 <string>Speed:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="variable_speed">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="statusTip">
                 <string>Speed value of the auto driver.</string>
                </property>
                <property name="suffix">
                 <string> rpm</string>
                </property>
                <property name="minimum">
                 <number>-100</number>
                </property>
                <property name="maximum">
                 <number>100</number>
                </property>
                <property name="singleStep">
                 <number>5</number>
                </property>
                <property name="value">
                 <number>-10</number>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QCheckBox" name="extremeRebound">
              <property name="statusTip">
               <string>When solver calls error, auto driver will change the direction.</string>
              </property>
              <property name="text">
               <string>Extreme rebound</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <widget class="QPushButton" name="variable_play">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="statusTip">
               <string>Start / Pause the auto driver of this variables.</string>
              </property>
              <property name="icon">
               <iconset resource="../../icons.qrc">
                <normaloff>:/icons/play.png</normaloff>
                <activeon>:/icons/pause.png</activeon>:/icons/play.png</iconset>
              </property>
              <property name="checkable">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="variable_stop">
              <property name="statusTip">
               <string>Stop the auto driver and return to original place.</string>
              </property>
              <property name="icon">
               <iconset resource="../../icons.qrc">
                <normaloff>:/icons/interrupted.png</normaloff>:/icons/interrupted.png</iconset>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <widget class="Line" name="line_5">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="record_groupBox">
     <property name="title">
      <string>Records</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_22">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <widget class="QLabel" name="record_interval_label">
          <property name="text">
           <string>Interval: </string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="record_interval">
          <property name="statusTip">
           <string>Each coordinate will be recorded after this angle value.</string>
          </property>
          <property name="suffix">
           <string>Â°</string>
          </property>
          <property name="minimum">
           <double>0.500000000000000</double>
          </property>
          <property name="maximum">
           <double>10.000000000000000</double>
          </property>
          <property name="value">
           <double>6.000000000000000</double>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QCheckBox" name="record_show">
          <property name="statusTip">
           <string>Show path data on the canvas.</string>
          </property>
          <property name="text">
           <string>Show path data</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_8">
        <item>
         <widget class="QListWidget" name="record_list">
          <property name="contextMenuPolicy">
           <enum>Qt::CustomContextMenu</enum>
          </property>
          <property name="statusTip">
           <string>All recorded path data of this workbook.</string>
          </property>
          <property name="currentRow">
           <number>0</number>
          </property>
          <item>
           <property name="text">
            <string>Auto preview</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QPushButton" name="record_start">
            <property name="statusTip">
             <string>Start / Stop record.</string>
            </property>
            <property name="icon">
             <iconset resource="../../icons.qrc">
              <normaloff>:/icons/record.png</normaloff>:/icons/record.png</iconset>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_6">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="record_remove">
            <property name="statusTip">
             <string>Delete the specified path data.</string>
            </property>
            <property name="icon">
             <iconset resource="../../icons.qrc">
              <normaloff>:/icons/delete.png</normaloff>:/icons/delete.png</iconset>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../../icons.qrc"/>
 </resources>
 <connections/>
</ui>
